plugin/metrics/factory.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/plugin"
+	"github.com/jaegertracing/jaeger/plugin/metrics/prometheus"
+	"github.com/jaegertracing/jaeger/storage"
+	"github.com/jaegertracing/jaeger/storage/metricsstore"
+)
+
+const (
+	prometheusStorageType = "prometheus"
+)
+
+// AllStorageTypes defines all available storage backends.
+var AllStorageTypes = []string{prometheusStorageType}
+
+// Factory implements storage.Factory interface as a meta-factory for storage components.
```

Comment:
Can you add a type-assertion, to make sure that this struct always implements the interface?